pages/getPreviewApiKey.ts

```diff
@@ -17,7 +17,7 @@ const GetPreviewApiKey: FC = () => {
     if (!path) {
       console.warn("Missing query parameter 'path' in /getPreviewApiKey. Will redirect to / after auth.");
     }
-    setCookie(urlAfterAuthCookieName, path);
+    setCookie(urlAfterAuthCookieName, path, { path: "/", secure: true, sameSite: "none" });
```

Comment:
we have a variable `cookieOptions` for that in `middleware.ts`. For me, it makes sense to extract it somewhere and reuse it. What do you think?